api_test.go

```
@@ -0,0 +1,206 @@
+	_addProduct(Product{Name: "coke", Description: "From Coke", Type: 1, CurrQuantity: 60, MinQuantity: 50})
+	_addProduct(Product{Name: "h2oh", Description: "From AmBev", Type: 2, CurrQuantity: 100, MinQuantity: 50})
+	_addProduct(Product{Name: "pepsi", Description: "From Pepsico", Type: 3, CurrQuantity: 10, MinQuantity: 20})
+	_addProduct(Product{Name: "original", Description: "From AmBev", Type: 4, CurrQuantity: 70, MinQuantity: 50})
+	_addProduct(Product{Name: "kuat", Description: "From Coke", Type: 5, CurrQuantity: 80, MinQuantity: 90})
+	_addProduct(Product{Name: "guarana", Description: "From From AmBev", Type: 6, CurrQuantity: 5, MinQuantity: 10})
+	_addProduct(Product{Name: "mate", Description: "From Sei la", Type: 7, CurrQuantity: 60, MinQuantity: 50})
+	_addProduct(Product{Name: "soda", Description: "From Coke company", Type: 8, CurrQuantity: 70, MinQuantity: 60})
+	_addProduct(Product{Name: "juice", Description: "From Mother Nature", Type: 9, CurrQuantity: 90, MinQuantity: 110})
+}
+
+func getBytes(p Product) []byte {
+	bjson, err := json.Marshal(p)
+	if err != nil {
+		panic(err)
+	}
+
+	return bjson
+}
+
+func init() {
+	_headers = make(map[string]string)
+	products = make(map[string]Product)
+	populateProducts()
+}
+
+func makeRequest(httpMethod string, url string, requestObj []byte, headers map[string]string) (*http.Response, error) {
+	req, err := http.NewRequest(httpMethod, url, bytes.NewBuffer(requestObj))
+	addHeaders(req, headers)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+
+}
+
+func addHeaders(req *http.Request, headers map[string]string) {
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+}
+
+// get - http://127.0.0.1:8080/api/inventory/product/:id
+func productExists(id int) bool {
+	response, err := makeRequest(router.GET, "http://127.0.0.1:8080/api/inventory/product/"+strconv.Itoa(id), make([]byte, 1), _headers)
+	if err != nil {
+		panic(err)
```

vc quer fail fast? Tipo, se uma request falhar vc para os testes?